zkevm-circuits/src/evm_circuit/execution/shl_shr.rs

```diff
@@ -0,0 +1,252 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        step::ExecutionState,
+        table::{FixedTableTag, Lookup},
+        util::{
+            self,
+            common_gadget::SameContextGadget,
+            constraint_builder::{ConstraintBuilder, StepStateTransition, Transition::Delta},
+            from_bytes,
+            math_gadget::{IsZeroGadget, LtWordGadget, MulAddWordsGadget},
+            sum, CachedRegion, Cell,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+use bus_mapping::evm::OpcodeId;
+use eth_types::{Field, ToLittleEndian, U256};
+use halo2_proofs::plonk::Error;
+
+/// ShlShrGadget verifies opcode SHL and SHR.
+/// For SHL, verify pop1 * (2^pop2) == push;
+/// For SHR, verify pop1 / (2^pop2) = push;
```

Comment:
```suggestion
/// For SHL, verify pop1 * (2^pop2) % 2^256 == push;
/// For SHR, verify pop1 / (2^pop2) % 2^256 == push;
```
This looks good otherwise.